### DISCUSSION TITLE
GPS smart usage by Detection Sensor, for battery conservation

### DISCUSSION THREAD

**Discussion by djoris** [meshtastic/firmware#5005] [Ideas]
GPS is a major energy consumer. Thus, it would be useful to enable the GPS chip opportunisticly.

A basic first option would be: when the detection sensor is triggered, start GPS. This could tie in with motion sensors, or the accelerometer (and lux!) sensors on the T1000-E.

There are many routes to make this even smarter, by tuning how often GPS should be polled during 'trigger' state.

1. Set a high interval 'in rest'. This will allow getting a position (and time fix!) at boot. It will also enable catching movement when somehow the detection sensors aren't triggered.
2. When GPS sees movement, this could trigger either detection sensor, or just extend the 'on the move' time.
3. When the triggers (sensors, movement) are set to false, the device could return to the 'in rest' state.

**Comment by Dylanliacc** [comment]
yeah！ let's try to do this

**Comment by djoris** [comment]
just to track progress: thanks also to @caveman99, the accelerometer on T1000-E is now supported!
https://github.com/meshtastic/firmware/pull/5141
